discovery/openstack/openstack.go

```diff
@@ -96,12 +98,18 @@ func (c *SDConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	if err != nil {
 		return err
 	}
+
+	if c.EndpointType != "public" && c.EndpointType != "admin" && c.EndpointType != "internal" {
```

Comment:
can you add a test case like in https://github.com/prometheus/prometheus/pull/4594/files?